src/com/legrig/CPSNonCPSTestScript.groovy

```diff
@@ -0,0 +1,53 @@
+package com.legrig
+
+import com.cloudbees.groovy.cps.NonCPS
+
+  @NonCPS
+  boolean nonCpsMethodCallingCpsMethod() {
+    boolean ret = false
+    println "this is a non cps method"
+    ret = cpsMethod()
+    return ret
+  }
+
+  /**
+   * This will always throw an exception
+   *
+   * @return
+   */
+  boolean cpsMethodCallingNonCpsCallingCps() {
+    boolean ret = false
+    println "this is a cps method calling non cps method"
+    nonCpsMethodCallingCpsMethod()
```

Comment:
should this be

ret=nonCpsMethodCallingCpsMethod()

Reply:
Yeah, good point, not that it matters - but it is good to be consistent :-)